superset/views/core.py

```diff
@@ -1221,10 +1221,14 @@ def explore(self, datasource_type=None, datasource_id=None):
             )
 
         standalone = request.args.get("standalone") == "true"
+        user_roles = [role.name.lower() for role in get_user_roles()]
+        is_admin_role = "admin" in user_roles
         bootstrap_data = {
             "can_add": slice_add_perm,
             "can_download": slice_download_perm,
             "can_overwrite": slice_overwrite_perm,
+            "can_overwrite_datasource": is_admin_role
```

Comment:
shouldn't this also be true for sql lab datasources?

Reply:
this bootstrap_data is only used by explore view.

Reply:
sorry, i meant datasources created by clicking the explore button in sql lab. those probably should be completely editable since they don't represent physical tables

Reply:
no. Even not associated physical table, this change still affect other chart that current using this datasource right?